File: Tema 03 - Web API/PetShelter/PetShelter.Domain/Fundraiser.cs
namespace PetShelter.Domain;

public class Fundraiser : INamedEntity
{
    public string Name { get; set; }
    public decimal GoalValue { get; set; }
    public decimal CurrentRaisedAmount { get; set; }
    public DateTime DueDate { get; set; }
    public DateTime CreationDate { get; set; }
    public FundraiserStatus Status { get; set; }
    public Person Owner { get; set; }
    public ICollection<Person> Donors { get; set; }
}

File: Tema 05 - Typescript/PetShelterFrontend/src/models/Fundraiser.ts
import { FundraiserPartialInfo } from "./FundraiserPartialInfo";
import type { Person } from "./Person"

export class Fundraiser extends FundraiserPartialInfo {
    currentRaisedAmount?: number;
    goalValue?: number;
    dueDate?: Date;
    donors: Person[] = [];

    constructor(name: string, status: string, currentRaisedAmount: number, goalValue: number, dueDate: Date, donors: Person[]) {
        super(name, status);
        this.name = name;
        this.status = status;
        this.currentRaisedAmount = currentRaisedAmount;
        this.goalValue = goalValue;
        this.dueDate = dueDate;
        this.donors = donors;
    }
}
